requests/models.py

```diff
@@ -403,10 +404,13 @@ def prepare_url(self, url, params):
     def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
+        self.headers = CaseInsensitiveDict()
         if headers:
-            self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
-        else:
-            self.headers = CaseInsensitiveDict()
+            pat = re.compile("^\S([^\r\n]+)?$")
```

Comment:
Good call on that, I hadn't thought out the regex completely. It was missing strings of length 0. Comment and test updated.